docs/source/torch.compiler_aot_inductor.rst

```diff
@@ -0,0 +1,230 @@
+    -- Detecting C compiler ABI info
+    -- Detecting C compiler ABI info - done
+    -- Check for working C compiler: /home/userid/local/ccache/lib/cc - skipped
+    -- Detecting C compile features
+    -- Detecting C compile features - done
+    -- Detecting CXX compiler ABI info
+    -- Detecting CXX compiler ABI info - done
+    -- Check for working CXX compiler: /home/userid/local/ccache/lib/c++ - skipped
+    -- Detecting CXX compile features
+    -- Detecting CXX compile features - done
+    -- Found CUDA: /usr/local/cuda-12.1 (found version "12.1")
+    -- The CUDA compiler identification is NVIDIA 12.1.105
+    -- Detecting CUDA compiler ABI info
+    -- Detecting CUDA compiler ABI info - done
+    -- Check for working CUDA compiler: /usr/local/cuda-12.1/bin/nvcc - skipped
+    -- Detecting CUDA compile features
+    -- Detecting CUDA compile features - done
+    -- Found CUDAToolkit: /usr/local/cuda-12.1/include (found version "12.1.105")
+    -- Looking for pthread.h
+    -- Looking for pthread.h - found
+    -- Performing Test CMAKE_HAVE_LIBC_PTHREAD
+    -- Performing Test CMAKE_HAVE_LIBC_PTHREAD - Success
+    -- Found Threads: TRUE
+    -- Caffe2: CUDA detected: 12.1
+    -- Caffe2: CUDA nvcc is: /home/userid/local/ccache/cuda/nvcc
+    -- Caffe2: CUDA toolkit directory: /usr/local/cuda-12.1
+    -- Caffe2: Header version is: 12.1
+    -- /usr/local/cuda-12.1/lib64/libnvrtc.so shorthash is b51b459d
+    -- USE_CUDNN is set to 0. Compiling without cuDNN support
+    -- USE_CUSPARSELT is set to 0. Compiling without cuSPARSELt support
+    -- Autodetected CUDA architecture(s):  8.0 8.0 8.0 8.0 8.0 8.0 8.0 8.0
+    -- Added CUDA NVCC flags for: -gencode;arch=compute_80,code=sm_80
+    -- Found Torch: /home/userid/local/miniconda3/envs/nightly/lib/python3.10/site-packages/torch/lib/libtorch.so
+    -- Configuring done
+    -- Generating done
+    -- Build files have been written to: /home/userid/local/aot_inductor_example/build
+
+    (nightly) [ ~/local/aot_inductor_example/build]$ cmake --build . --config Release
+    [ 33%] Generating model_so_path.txt
+    Compiled model into: /tmp/torchinductor_userid/csnavcwn65mvhieu3jsqd2xkbzynhq2qif7rthy5l57qca3e4wwe/c64ucf56t5hvtulhodvu47apn2rcdhjre7ifghsuwovvniggmwd7.so
+    [ 66%] Building CXX object CMakeFiles/aot_inductor_example.dir/inference.cpp.o
+    [100%] Linking CXX executable aot_inductor_example
+    [100%] Built target aot_inductor_example
+
+After the ``aot_inductor_example`` binary has been generated in the ``build`` directory, executing it will
+display results akin to the following:
+
+.. code-block:: shell
+
+    (nightly) [ ~/local/aot_inductor_example/build]$ ./aot_inductor_example
+    Result from first inference:
+    0.4866
+    0.5184
+    0.4462
+    0.4611
+    0.4744
+    0.4811
+    0.4938
+    0.4193
+    [ CUDAFloatType{8,1} ]
+    Result from second inference:
+    0.4883
+    0.4703
+    [ CUDAFloatType{2,1} ]
```

Comment:
do you wanna end by linking to the python wrapper as well?

Reply:
By python wrapper, do you mean loading the compiled shared library using pybind?

Reply:
I would agree that the best way to reduce confusion here is to list all options and explain when to use which.
But maybe as a one-paragraph intro at the top.